flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -72,6 +72,9 @@
 	/** Determines if a tasks are failed or not if there is an error in their checkpointing. Default: true */
 	private boolean failOnCheckpointingErrors = true;
 
+	/** Determines if a job will fallback to checkpoint when there is a savepoint later than checkpoint. **/
+	private boolean perferCheckpointForRecovery = false;
```

Comment:
I am personally not a big fan of configuring checkpoint parameters in the CheckpointConfig class, I actually wish that all these options would be exposed as general flink configuration parameters (settable in flink-conf.yaml).

What do you think @StefanRRichter ?

Reply:
I think this config option belongs to checkpoint field. So provide it in `CheckpointConfig` sounds more reasonable. What's more, it is a job level config option, right? However, `flink-conf.yaml` is a cluster-level configuration file(if users deploy Flink with standalone mode, it will affect all the jobs).